zilliqa/src/consensus.rs

```diff
@@ -1903,37 +1953,80 @@ impl Consensus {
     }
 
     pub fn leader_at_block(&self, block: &Block, view: u64) -> Option<Validator> {
+        if let Some((cached_block_num, cached_view, leader)) = *self.leader_cache.borrow() {
+            if cached_block_num == block.number() && cached_view == view {
+                return Some(leader);
+            }
+        }
```

Comment:
Is it intentional that we don't update the cache here (and only inside `proposal()`)?
If so I'd consider renaming it to something like `latest_leader_cache`, since my first assumption upon seeing the field called `leader_cache` is that it would be used to store whatever the last accessed leader was, rather than just specifically the leader from the latest proposal.